junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/HandlerCallCounter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2015-2024 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+class HandlerCallCounter {
```

Comment:
As this class isn't used outside of `LifecycleMethodExecutionExceptionHandlerTests` it can be an inner class. This keeps the test contained to its own file, removing the need to consider the counter outside of the context in which it is used.